### PR TITLE
Added grading run log viewing for staff.

### LINKED ISSUE #34: View Logs
We should be able to view the logs for an assignment through a staff-protected web page. Dependent issue in https://github.com/illinois-cs241/broadway-api/issues/46

### PR DESCRIPTION
**Pull request by shreyas208**
Resolves #34.

### CODE REVIEW

File: src/bw_api.py [hunk 1]
```diff
@@ -48,11 +48,57 @@ def get_grading_run_status(cid, aid, run_id):
...
+	except KeyError as e:
+		logging.error("get_grading_job_log(cid={}, job_id={}): {}".format(cid, job_id, repr(e)))
+		return None
+	except JSONDecodeError as e:
+		logging.error("get_grading_job_log(cid={}, job_id={}): {}".format(cid, job_id, repr(e)))
+		return None
+
+
+def get_grading_run_log(cid, run_id):
+	return {"stdout": "Lorem ipsum dolor sit amet.\nLorem ipsum? Dolor! Sit amet.", "stderr": "Lorem ipsum dolor sit amet.\nLorem ipsum? Dolor! Sit amet."}
```
Comment (rod-lin) [inline src/bw_api.py]: Umm is this line for testing?

File: src/bw_api.py [hunk 2]
```diff
@@ -48,11 +48,57 @@ def get_grading_run_status(cid, aid, run_id):
 	:return: a status string if successful, or None otherwise.
 	"""
 	try:
-		resp = requests.get(url="%s/grading_run/%s/%s/%s" % (BROADWAY_API_URL, cid, aid, run_id), headers=HEADERS)
-		return resp.json()["data"]["student_jobs_state"][0]
-	except requests.exceptions.RequestException:
+		resp = requests.get(url="%s/grading_run_status/%s/%s" % (BROADWAY_API_URL, cid, run_id), headers=HEADERS)
+		student_jobs_dict = resp.json()["data"]["student_jobs"]
+		return student_jobs_dict.values().next()
```
Comment (rod-lin) [inline src/bw_api.py]: I don't recall there is a `next` method for `dict_values`. Is this method added by the json library or something?

File: src/bw_api.py (hunk 1 above)
Comment (shreyas208) [reply src/bw_api.py]: Whoops. Good catch :no_mouth:

File: src/bw_api.py (hunk 2 above)
Comment (shreyas208) [reply src/bw_api.py]: `dict.values()` should return an iterator in Python3, so `next()` should return the first item in the iterator.

**Comment by shreyas208** [comment]
### Action buttons
image

### Log modal
image

**Comment by redsn0w422** [comment]
OOC, is the standard out/err going to have test output, or explicit prints made in the testers?

File: src/bw_api.py (hunk 2 above)
Comment (shreyas208) [reply src/bw_api.py]: Nvm, you're right. It's not an iterator.

**Comment by nmagerko** [comment]
@redsn0w422 it'll just be the logs directly out of the container, so basically whatever you'd see if you ran the container(s) locally.

@shreyas208 the logs are usually pretty verbose. You might want to consider having the portions that actually contain the stderr/stdout text be scollable after a certain point so that things don't get out of control. Looking real good though.

**Comment by ayushr2** [comment]
Broadway side of things LGTM. illinois-cs241/broadway-on-demand#46 has been resolved so we can try this in production.

**Pull request merged**